### DISCUSSION TITLE
What is a type?

### DISCUSSION THREAD

**Discussion by andrew222651** [python/typing#923] [Q&A] [answered]
In Python how do you tell if an object is a type? Are types well defined? Is an instance of `TypeVar` a type?

**Comment by JelleZijlstra** [comment, marked as answer]
"Type" can have two different plausible meanings:
- A runtime type object, such as a user-defined class or a builtin type like `int`. At runtime you can check for these with `isinstance(obj, type)`. In the type system they are represented as e.g. `type[int]` (or `Type[int]` in older versions). An instance of TypeVar is not a type in this definition.
- Something that is valid as an annotation in the type system. That includes all runtime types, but also some other objects, such as TypeVars and special forms like Any. There is no runtime function that tells you whether something is valid as a type annotation.

**Reply by sobolevn** [reply]
> There is no runtime function that tells you whether something is valid as a type annotation.

However, there are some 3rd party runtime type checkers that can do that. List: https://github.com/typeddjango/awesome-python-typing/#dynamic-type-checkers
